app/src/main/java/cm/aptoide/pt/wallet/WalletInstallPresenter.kt

```diff
@@ -1,22 +1,50 @@
 package cm.aptoide.pt.wallet
 
 import android.os.Build
+import cm.aptoide.pt.actions.PermissionManager
+import cm.aptoide.pt.actions.PermissionService
 import cm.aptoide.pt.presenter.Presenter
 import cm.aptoide.pt.presenter.View
-import cm.aptoide.pt.promotions.PromotionsManager
 import cm.aptoide.pt.promotions.WalletApp
+import rx.Completable
 import rx.Observable
 import rx.Scheduler
+import rx.schedulers.Schedulers
 
 class WalletInstallPresenter(val view: WalletInstallView,
                              val walletInstallManager: WalletInstallManager,
                              val navigator: WalletInstallNavigator,
-                             val promotionsManager: PromotionsManager,
-                             val viewScheduler: Scheduler) : Presenter {
+                             val permissionManager: PermissionManager,
+                             val permissionService: PermissionService,
+                             val viewScheduler: Scheduler, val io: Scheduler) : Presenter {
 
   override fun present() {
     loadWalletInstall()
     handleCloseButtonClick()
+    handleCancelDownloadButton()
+  }
+
+  private fun handleCancelDownloadButton() {
+    view.lifecycleEvent
+        .filter { lifecycleEvent -> View.LifecycleEvent.CREATE == lifecycleEvent }
+        .flatMap { view.cancelDownloadButtonClicked() }
+        .flatMap { walletInstallManager.getWallet() }.first()
```

Comment:
yes, because it is an observable and in this case we don't want to keep observing the download. We only got the app here, just because (like we are doing in appview) we want to have the wallet object to be able to call the remove method.